src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,29 @@
 'use strict';
 
 class Animal {
-  // write your code here
+  constructor(name, health = 100) {
+    this.name = name;
+    this.health = health;
+    this.hidden = false;
+    Animal.alive.push(this);
+  };
 }
 
+Animal.alive = [];
+
 class Herbivore extends Animal {
-  // write your code here
+  hide() {
+    this.hidden = !this.hidden;
+  }
 }
 
 class Carnivore extends Animal {
-  // write your code here
+  bite(herbivore) {
+    if (!herbivore.hidden && !(herbivore instanceof Carnivore)) {
+      herbivore.health -= 50;
+      Animal.alive.splice(Animal.alive.indexOf(herbivore), 1);
+    }
```

Comment:
```suggestion
  bite(anybody) {
    if (!anybody.hidden && !(anybody instanceof Carnivore)) {
      anybody.health -= 50;
      Animal.alive.splice(Animal.alive.indexOf(anybody), 1);
    }
```
You can bite any animals.(But not Carnivore(22 row))
I don't know, how tests passed, because I see that - 23 row you bitting the anybody and next step delete it in ```Aminal.alive```. If use ```filter``` by ```health``` it will be better.